### PR TITLE
Added support in asset.fetch as locations can be single url

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Added support in asset.fetch as locations can be single url
and that case it can be string url

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

File: avocado/utils/asset.py [hunk 1]
```diff
@@ -169,8 +170,11 @@ def fetch(self):
         # Now we have a writable cache_dir. Let's get the asset.
         # Adding the user defined locations to the urls list:
         if self.locations is not None:
-            for item in self.locations:
-                urls.append(item)
+            if isinstance(self.locations, list):
+                for item in self.locations:
+                    urls.append(item)
+            else:
+                urls.append(self.locations)
```
Comment (apahim) [inline avocado/utils/asset.py]: What if it's a tuple? :)
The beauty of the current implementation is that it works for any kind of sequence or iterator (lists, tuples, dictionaries, sets, custom iterators, ...).
The only exception is your case: a string. So, you should handle your case without breaking anything else:
```python
    def __init__(...):
        ...
        if isinstance(locations, str):
            self.locations = [locations]
        else:
            self.locations = locations
```
The rest stays the same.

Comment (PraveenPenguin) [reply avocado/utils/asset.py]: @apahim ah  yes right make sense , will do that

**Comment by PraveenPenguin** [comment]
@apahim pushed the changes please have a look

File: avocado/utils/asset.py [hunk 2]
```diff
@@ -65,7 +65,8 @@ def __init__(self, name, asset_hash, algorithm, locations, cache_dirs,
         :param name: the asset filename. url is also supported
         :param asset_hash: asset hash
         :param algorithm: hash algorithm
-        :param locations: list of locations where the asset can be fetched from
+        :param locations: list of locations/single location Url as String
+                          where the asset can be fetched from
```
Comment (apahim) [inline avocado/utils/asset.py]: Shall we rephrase this? We don't mention types for the other arguments... we can also omit that here now. Something like:
```
:param locations: location(s) where the asset can be fetched from
```
Considering that we now support basically anything, that's accurate and consistent.

Comment (PraveenPenguin) [reply avocado/utils/asset.py]: yes sure that right

**Comment by PraveenPenguin** [comment]
@apahim adopted the suggested review comment , please have look

**Pull request merged**